edx_proctoring/views.py

```diff
@@ -496,6 +499,61 @@ def get(self, request):
         elif relevant_attempt:
             data['onboarding_status'] = relevant_attempt.status
 
+        if waffle.switch_is_active(ONBOARDING_PROFILE_API):
+            attempt_data = None
+            try:
+                attempt_data = backend.get_onboarding_profile_info(course_id=course_id, user_id=user.id)
+            except BackendProviderOnboardingProfilesException as exc:
+                log_message = (
+                    'Failed to use backend onboarding status API endpoint for user_id={user_id}'
+                    'in course_id={course_id} because backend failed to respond. '
+                    'Status: {status}, Response: {response}'.format(
+                        user_id=user.id,
+                        course_id=course_id,
+                        response=str(exc),
+                        status=exc.http_status,
+                    )
+                )
+                LOG.warning(log_message)
+                return Response(data)
```

Comment:
I understand this data is coming from the logic before. Can we package that logic and the logic into a new function so we can understand the boundary of the onboarding status from attempts logic better?

Reply:
I just moved the logic that gets attempt data from our models to a function in api.py, this will make cleanup work for the waffle flag much easier